migrating_from_ocp_3_to_4/planning-migration-3-4.adoc

```diff
@@ -164,7 +164,10 @@ For more information, see xref:../networking/network_security/network_policy/abo
 
 In {product-title} 3.11, OpenShift SDN was the default networking plugin in Red Hat OpenShift Networking. In {product-title} {product-version}, OVN-Kubernetes is now the default networking plugin.
 
-For information on migrating to OVN-Kubernetes from OpenShift SDN, see xref:../networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.adoc#migrate-from-openshift-sdn[Migrating from the OpenShift SDN network plugin].
+[WARNING]
+====
+No supported method exists for migrating from the OpenShift SDN network plugin to the OVN-Kubernetes network plugin in {product-title} 4.17. You must migrate to the OVN-Kubernetes plugin before upgrading to {product-title} 4.17.
+====
```

Comment:
You can't migrate from sdn to ovn-k in OCP 3 though, so there's no way to do it before upgrading.

This document already says earlier "`It is not possible to upgrade your existing OpenShift Container Platform 3 cluster to OpenShift Container Platform 4. You must start with a new OpenShift Container Platform 4 installation.`", so I think the link to the migration documentation must have been just because that was where the differences between OpenShift SDN and OVN Kubernetes were documented...

I think the big points are:
- ovn-k doesn't have anything like openshift-sdn Multitenant mode; it always uses NetworkPolicy. (Maybe link to https://docs.openshift.com/container-platform/4.15/networking/network_policy/multitenant-network-policy.html as an alternative)
- ovn-k has egress IPs, egress firewall, egress routers, and multicast, but all of them are configured differently than they are in openshift-sdn (with links to the relevant pages).
- ovn-k uses a few private internal networks and you may need to edit the configuration if the default IP ranges for those networks overlap your networks.

Reply:
I _think_ the section is trying to inform users about the differences as you pointed out, which is possibly why we linked to the migration docs. It doesn't make sense, IMO to point to netpol docs here. We should point to OVN-K docs so users can read up on what OVN-K.

Reply:
Yes, we should tell them that they should install ovn-k from the start in their 4.x cluster.
(I'm not sure why you're saying 4.12? they should install whatever version is current at the time they are switching to 4.x.)

> It doesn't make sense, IMO to point to netpol docs here.

If the customer is using openshift-sdn in multitenant mode in 3.x (which was the default), they will want to know how to configure multitenant-like functionality in 4.x. Right now, the best answer for that is the page about implementing multitenancy with NetworkPolicies, though eventually the right answer will be "use AdminNetworkPolicy" (once ANP gets tenancy support).

> We should point to OVN-K docs so users can read up on what OVN-K is.

Right, but the general ovn-k docs talk about _everything_. It may be worth calling out the handful of features that also existed in openshift-sdn in 3.x (egress IPs, egress firewall, egress router, multicast) since they will be of particular interest to this group of users.

Reply:
We put some thought into why OVN-K is the better choice in this KB too: https://access.redhat.com/articles/7065170